### PR TITLE
Datadog: Log request headers

### PR DESCRIPTION
**Pull request by prabhanshuguptagit**
**Story card:** -
## Because

We need to peek at the request headers in Datadog which aren't logged (tagged) anywhere by default.

## This addresses

This configures datadog's rack middleware to log the custom headers we use. [ref](https://docs.datadoghq.com/tracing/setup/ruby/#rack)

### CODE REVIEW

File: config/initializers/00_datadog.rb [hunk 1]
```diff
@@ -10,6 +10,7 @@
   c.tracer.enabled = SEND_DATA_TO_DD_AGENT
   c.version = SimpleServer.git_ref(short: true)
   c.use :rails, analytics_enabled: true
+  c.use :rack, headers: {request: ["X-USER-ID", "X-FACILITY-ID", "X-SYNC-GROUP-ID"], response: ["Content-Type", "X-Request-ID"]}
```
Comment (harimohanraj89) [inline config/initializers/00_datadog.rb:13]: Is the `X-Request-ID` set by DataDog?

PS does the capitalization have to be inconsistent between request and response headers?

**Review by harimohanraj89 — APPROVED** [review]
Couple non-blocking questions.

Let's test and :shipit:

File: config/initializers/00_datadog.rb (hunk 1 above)
Comment (prabhanshuguptagit) [reply config/initializers/00_datadog.rb:13]: The `Request ID` is set either by rails or nginx, not sure which one. This was already being logged in datadog as a [default](https://github.com/DataDog/dd-trace-rb/pull/389)

The headers are case insensitive, I just stuck to what they are conventionally being called around everywhere.

Comment (harimohanraj89) [reply config/initializers/00_datadog.rb:13]: I guess that's fair. You could also camel(?)case em all if you want. Whichever you think is prettier.

**Pull request merged**